frontends/api/src/hooks/learningResources/keyFactory.ts

```diff
@@ -30,8 +30,18 @@ import type {
   UserListRelationship,
   MicroUserListRelationship,
 } from "../../generated/v1"
+
 import { createQueryKeys } from "@lukemorales/query-key-factory"
 
+const shuffle = ([...arr]) => {
+  let m = arr.length
+  while (m) {
+    const i = Math.floor(Math.random() * m--)
+    ;[arr[m], arr[i]] = [arr[i], arr[m]]
+  }
+  return arr
+}
+
```

Comment:
The backend randomizer used to randomize results within each learning path position index, so for example, assuming 5 "featured resource" learning paths:

Carousel positions 1-5 would be the 1st course from each of the 5 paths, in random order
Carousel positions 6-10 would be the 2nd course from each of the 5 paths, in random order
etc.

This ensured that all units would be equally represented, with the top courses appearing first.  

This PR seems to make the order completely random, for instance the first time I tried it I got:
1. Position #3 course from MITx
2. Position #1 course from Professional Education
3. Position #1 course from OCW
4. Position #2 course from Sloan
5. Position #3 course from Sloan
etc

Have the specs changed so that the order can be completely random now, or should the old pseudo-random sorting still be in place?

Reply:
I just asked Ferdi - he said try to preserve the old randomization if its easy enough so I'm going to refactor this to match what it was previously there.